File: bamboo/lib/bamboo/member.ex
defmodule Bamboo.Member do
  use Ecto.Schema
  import Ecto.Changeset

  schema "members" do
    field :sn, :string
    field :name, :string
    field :memo, :string
    field :wifi_password, :string
    field :version, :integer
    field :deleted_at, :utc_datetime
    has_many :hosts, Bamboo.Host

    timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(member, attrs) do
    member
    |> cast(attrs, [:sn, :name, :memo, :wifi_password, :version, :deleted_at])
    |> validate_required([:sn, :name, :memo, :wifi_password, :version])
  end
end

File: loquat/models/member.go
package models

import "gorm.io/gorm"

type Member struct {
	gorm.Model

	Sn           string `gorm:"uniqueIndex;not null;size:15"`
	Name         string `gorm:"index;not null;size:31"`
	WifiPassword string `gorm:"not null;size:255"`
	Memo         string `gorm:"not null;type:text"`
	Version      uint   `gorm:"not null;default:0"`
	Hosts        []Host
}

func (Member) TableName() string {
	return "members"
}
